### DISCUSSION TITLE
Reduce firmware size

### DISCUSSION THREAD

**Discussion by AndreoBotelho** [lvgl-micropython/lvgl_micropython#103] [Board Specific]
I'm building for a STM32 board with only the internal 1MB flash storage and even with LTO=1 and only 14px builtin font the firmware takes all the storage, (micropython firmware takes half the size), what can I disable to reduce the firmware size taken by lvgl? I need at least 128kb of flash to user storage.

**Comment by kdschlosser** [comment]
it is never going to fit on an MCU that has only 1MB of flash. It's an impossibility. MicroPython all by it's self is going to consume at least 500K of that. and LVGL at the absolute smallest is can possibly be is going to be something around 1.3 to 1.4 MB. The smallest I have managed to get the compiled size down to is 1.8mb and that is with turning off  every thing that is able to be turned off in LVGL.

**Reply by kdschlosser** [reply]
You can give it a shot by editing `lib/lv_conf.h` and turning off the widgets and any other "extras" that might be turned on. The generated binding code is massive in size the source file is around 1.6mb in size with the most common features turned on.

**Reply by AndreoBotelho** [reply]
with default -Os the firmware have a little more than 1Mb, using LTO option I managed to get around 980kb and removing the fonts it gets 930kb so I could use a 64k flash storage, really low but useful to test, now I could compile em test lvgl but I'll have to use a SPI flash storage or SD card to run applications

**Comment by kdschlosser** [comment]
I have to fix the stm32 build because it is not working at the moment. I made some changes to the SPI class that will allow the sharing of the SPI bus with the SDCard and also with touch drivers. This was not possible to do with the way it was written originally because of how the SPI works for the display. I have to finish making the modifications to the STM32 port to get it to work...

**Comment by kdschlosser** [comment]
have you turned off all of the extras in MicroPython. things like bluetooth and wifi and whatever other things are turned on that you may not use...
